compiler/damlc/daml-ide-core/test/Development/IDE/State/API/Testing.hs

```diff
@@ -491,6 +509,29 @@ timedSection targetDiffTime block = do
         throwError $ TimedSectionTookTooLong targetDiffTime actualDiffTime
     return value
 
+
+templateChoicesToProps :: TemplateChoices -> TemplateProp
+templateChoicesToProps tca = TemplateProp choicesInTpl (sum $ map (length . actions ) (choiceAndActions tca))
+    where choicesInTpl = Set.fromList $ map (\ca -> ExpectedChoices ( DAP.renderPretty $ choiceName ca) (choiceConsuming ca)) (choiceAndActions tca)
+
+graphTest :: LF.World -> LF.Package -> Set.Set TemplateProp -> Either [TemplateProp] ()
+graphTest wrld lfPkg expectedProps =
+    if expectedProps == (Set.fromList $ map templateChoicesToProps tplPropsActual)
```

Comment:
I would suggest that you move this to `ShakeTest`. That way you can slightly simply things by not having to pattern match on the result at the callsite.
```haskell
graphTest :: LF.World -> LF.Package -> Set.Set TemplateProp -> ShakeTest ()
graphTest wrld lfPkg expectedProps = do
    let actual = Set.fromList $ map templateChoicesToProps tplPropsActual
    unless expectedProps == actual $
        throwError $ ExpectedTemplateProps expectedProps actual

Reply:
cool. That looks better. Done